### PR TITLE
CURLOPT_HTTP_VERSION

### PR DESCRIPTION
**Pull request by kni**
To select HTTP version one should use $req->protocol("HTTP/1.1");

### CODE REVIEW

File: lib/LWP/Protocol/Net/Curl.pm [hunk 1]
```diff
@@ -217,6 +217,18 @@ sub _handle_method {
         },
     );
 
+	if (my $protocol = $request->protocol) {
+		my $v;
+		if ($protocol =~ m/^HTTP\/1\.0$/) {
+			$v = CURL_HTTP_VERSION_1_0;
+		} elsif ($protocol =~ m/^HTTP\/1\.1$/) {
+			$v = CURL_HTTP_VERSION_1_1;
+		} elsif ($protocol =~ m/^HTTP\/2$/) {
```
Comment (creaktive) [inline lib/LWP/Protocol/Net/Curl.pm]: Are you sure it's always `HTTP/2` and never `HTTP/2.0`? The RFC has at least one occurrence of the latter: https://tools.ietf.org/html/rfc7540#section-3.5

Comment (creaktive) [reply lib/LWP/Protocol/Net/Curl.pm]: Also, libcurl is much more granular regarding the version setting; it even has HTTP/3: https://github.com/curl/curl/blob/master/docs/libcurl/symbols-in-versions#L832-L839

File: lib/LWP/Protocol/Net/Curl.pm [hunk 2]
```diff
@@ -217,6 +217,18 @@ sub _handle_method {
...
+			$v = CURL_HTTP_VERSION_2TLS;
```
Comment (creaktive) [inline lib/LWP/Protocol/Net/Curl.pm]: Please note that the `CURL_HTTP_VERSION_2TLS` symbol does not exist in libcurl prior to 7.47.0. It is worth checking within an `eval` if it is available at all. On the other hand, such check does not need to happen on every `_handle_method` call; it should be done during the module load.
Sorry for being so picky!

**Comment by kni** [comment]
Hello. 
I add CURL_HTTP_VERSION_3 and the eval (but in _handle_method). 
Do the eval is so heavy?

Or it is the better to do as:
```
my $http_version_2 = eval { Net::Curl::Easy::CURL_HTTP_VERSION_2TLS() };
my $http_version_3 = eval { Net::Curl::Easy::CURL_HTTP_VERSION_3() };
# ....
		} elsif ($protocol =~ m/^HTTP\/2/) {
			$v =  $http_version_2;
		} elsif ($protocol =~ m/^HTTP\/3/) {
			$v =  $http_version_3;
```

**Comment by creaktive** [comment]
In practice it makes very little difference if you run the evals during the module load or for every request :)
Thanks for the PR, I’ll release to CPAN on Monday!

**Comment by kni** [comment]
Thank you very much!

**Pull request merged**